packages/node/src/generators/setup-docker/setup-docker.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  addDependenciesToPackageJson,
+  convertNxGenerator,
+  formatFiles,
+  generateFiles,
+  GeneratorCallback,
+  joinPathFragments,
+  logger,
+  readProjectConfiguration,
+  Tree,
+  updateProjectConfiguration,
+} from '@nrwl/devkit';
+import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
+import { writeFileSync } from 'fs-extra';
+import { SetUpDockerOptions } from './schema';
+
+function normalizeOptions(
+  setupOptions: SetUpDockerOptions
+): SetUpDockerOptions {
+  return {
+    ...setupOptions,
+    buildTarget: setupOptions.buildTarget ?? 'build',
+  };
+}
+
+function addDocker(tree: Tree, options: SetUpDockerOptions) {
+  const project = readProjectConfiguration(tree, options.projectName);
+  if (!project || !options.targetName) {
+    return;
+  }
+
+  if (tree.exists(joinPathFragments(project.root, 'DockerFile'))) {
+    logger.info(
+      `Skipping setup since a DockFile already exists inside ${project.root}`
+    );
+  }
+  const outputPath =
+    project.targets[`${options.buildTarget}`]?.options.outputPath;
+  generateFiles(tree, joinPathFragments(__dirname, './files'), project.root, {
+    tmpl: '',
+    app: project.sourceRoot,
+    buildLocation: outputPath,
+  });
+}
+
+export function updateProjectConfig(tree: Tree, options: SetUpDockerOptions) {
+  let projectConfig = readProjectConfiguration(tree, options.projectName);
+
+  const buildPort = process.env.PORT ?? 3000;
+
+  projectConfig.targets[`${options.targetName}`] = {
```

Comment:
Consider adding a `production` config ?

See https://github.com/nrwl/nx-console/issues/1466

Reply:
This will probably be a follow-up since usually we have both `development` and `production` configs for most of our projects.
It might be useful to add this for all Node projects and not just `docker`